File: src/main/java/net/treset/minecraftlauncher/util/ui/sort/LauncherModDisabledNameComparator.java
package net.treset.minecraftlauncher.util.ui.sort;

import net.treset.mc_version_loader.launcher.LauncherMod;
import net.treset.minecraftlauncher.LauncherApplication;

import java.util.Comparator;

public class LauncherModDisabledNameComparator implements Comparator<LauncherMod> {
    @Override
    public int compare(LauncherMod o1, LauncherMod o2) {
        if(o1.isEnabled() == o2.isEnabled()) {
            return o1.getName().compareTo(o2.getName());
        }
        if(o1.isEnabled()) {
            return -1;
        }
        return 1;
    }

    @Override
    public String toString() {
        if(LauncherApplication.stringLocalizer == null) {
            return "sort.disabled.name";
        }
        return LauncherApplication.stringLocalizer.get("sort.disabled.name");
    }
}

File: client/composeApp/src/commonMain/kotlin/net/treset/treelauncher/backend/util/sort/LauncherModDisabledNameComparator.kt
package net.treset.treelauncher.backend.util.sort

import net.treset.treelauncher.backend.data.LauncherMod
import net.treset.treelauncher.localization.strings

class LauncherModDisabledNameComparator : Comparator<LauncherMod> {
    override fun compare(o1: LauncherMod, o2: LauncherMod): Int {
        if (o1.isEnabled == o2.isEnabled) {
            return o1.name.lowercase().compareTo(o2.name.lowercase())
        }
        return if (o1.isEnabled) {
            -1
        } else 1
    }

    override fun toString(): String = strings().sortBox.sort.enabledName()
}
